core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java

```diff
@@ -50,12 +55,34 @@ private void updateScore(
                 node.setProperty(ContentScoreUpdateServiceImpl.CONTENT_SCORE_UPDATED, now);
                 node.setProperty("cq:lastModified", now);
                 node.setProperty("cq:lastModifiedBy", "scale-of-belief");
-                Session session = resourceResolver.adaptTo(Session.class);
 
-                if (session != null) {
-                    session.save();
-                }
+                TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+                List<Tag> newTags = buildTagsWithScore(contentResource, tagManager, score);
+                tagManager.setTags(contentResource, newTags.toArray(new Tag[0]));
+            }
+        }
+    }
+
+    private List<Tag> buildTagsWithScore(
+        final Resource contentResource,
+        final TagManager tagManager,
+        final int score) {
+
+        Tag[] existingTags = tagManager.getTags(contentResource);
+        List<Tag> newTags = Lists.newArrayList();
+
+        for (Tag existingTag : existingTags) {
+            if (existingTag.getTagID().startsWith(SCALE_OF_BELIEF_TAG_PREFIX)) {
+                continue;
             }
+            newTags.add(existingTag);
         }
+
+        Tag scoreTag = tagManager.resolve(SCALE_OF_BELIEF_TAG_PREFIX + score);
```

Comment:
someone/something needs to create these tags initially, right? is that going to be done manually?

Reply:
The tags already exist, and I think it was done manually, yes.